src/lexer.coffee

```diff
@@ -1113,11 +1113,7 @@ exports.isUnassignable = isUnassignable
 # “sometimes” keyword.
 isForFrom = (prev) ->
   if prev[0] is 'IDENTIFIER'
-    # `for i from from`, `for from from iterable`
-    if prev[1] is 'from'
-      prev[1][0] = 'IDENTIFIER'
```

Comment:
@helixbass This was throwing in Module mode only, because it tried to assign `'IDENTIFIER'` to a character in a string: `prev[1]` is the string `'from'`, so `prev[1][0]` is the character `f` in that string. In Script mode this just silently fails, whereas in Module it throws.

It appears this isn’t needed at all, as removing it doesn’t break any tests. I don’t remember what I was thinking. Does this all seem correct to you?

Reply:
@GeoffreyBooth it does make sense that assigning to `prev[1][0]` wouldn't have been correct

Tracing the possibilities from the comment, it also looks correct that this logic wasn't needed:

`for from from iterable` is effectively already covered by the `else if prev[0] is 'FOR'` clause below (as implied by its comment ``# `for from…` ``

`for i from from` won't actually make it this far when checking its second `from` (`isForFrom()` won't get called) since `@seenFor` will have been set to `no` when the first `from` is checked

So I'd recommend just updating and moving the comment:
```
  # `for i from iterable`
  if prev[0] is 'IDENTIFIER'
    yes
  # `for from…`
  ...
```

Reply:
Aside from the placement of the comment, I think what I have now matches what you’re suggesting?

Reply:
I was also suggesting removing  `` `for i from from` `` and `` `for from from iterable` `` from the comment since they're not actually covered by this case